packages/openid4vc/src/openid4vc-holder/OpenId4vpHolderServiceOptions.ts

```diff
@@ -1,20 +1,32 @@
 import type {
+  CanBePromise,
   DcqlCredentialsForRequest,
   DcqlQueryResult,
   DifPexCredentialsForRequest,
   DifPexInputDescriptorToCredentials,
   DifPresentationExchangeDefinition,
   EncodedX509Certificate,
+  JwsService,
 } from '@credo-ts/core'
-import { ResolvedOpenid4vpAuthorizationRequest } from '@openid4vc/openid4vp'
-import type { OpenId4VpAuthorizationRequestPayload } from '../shared'
+import { Openid4vpAuthorizationRequestDcApi, ResolvedOpenid4vpAuthorizationRequest } from '@openid4vc/openid4vp'
+import type { OpenId4VpAuthorizationRequestPayload, Openid4vpAuthorizationRequest } from '../shared'
 
 // TODO: export from oid4vp
 export type ParsedTransactionDataEntry = NonNullable<ResolvedOpenid4vpAuthorizationRequest['transactionData']>[number]
 
+export type VerifyAuthorizationRequestOptions = {
+  authorizationRequest: Openid4vpAuthorizationRequest | Openid4vpAuthorizationRequestDcApi
+  jwsService: JwsService
```

Comment:
I think it makes more sense to pass the `agentContext` as the first param to the callback. The user can then inject anything they need. Attestations could be other things than JWTs as well, so it's too focused on a specific case.

If the context is passed the user can just do `agentContext.resolve(JwsService)`

Reply:
Yes, makes sense